media/docs/cpp/cute/xe_2d_copy.md

```diff
@@ -0,0 +1,136 @@
+| `LD` / `ST`          | `LD` = Load from global memory; `ST` = Store to global memory             |
+| `N`, `T`, `V`        | Memory layout: `N` (row-major), `T` (column-major), `V` (VNNI, row-major) |
+
+Note: DataType does not represent the actual type of the data being loaded, but rather the width of the data type.
+For example, when loading an array of FP8 values a U8 copy operation should be used. 
+
+---
+
+## 2D Copy Execution Model
+
+All copy operations are performed at the **subgroup** level. Each **subgroup consists of 16 work-items**, 
+and these **work-items** cooperate to move a 2D block of data.
+
+### Data Distribution (Unpacked Copies)
+
+For a load like `XE_2D_U16x32x16_LD_N`, the 32 rows by 16 columns block of elements is split such that:
+
+- Each **work-item gets one column** with **32 elements**
+
+#### Example:
+| Work-item | Columns Loaded |
+|-----------|----------------|
+| 0         | Column 0       |
+| 1         | Column 1       |
+| ...       | ...            |
+| 15        | Column 15      |
+
+If we increase BlockWidth, for example using a `XE_2D_U16x32x32_LD_N` copy,
+it will load a 32 rows by 32 columns block of elements such that:
+
+- Each **work-item gets two columns**, each with **32 elements**
+- The columns are **16 apart**
+
+#### Example:
+| Work-item | Columns Loaded          |
+|-----------|-------------------------|
+| 0         | Column 0 and Column 16  |
+| 1         | Column 1 and Column 17  |
+| ...       | ...                     |
+| 15        | Column 15 and Column 31 |
+
+The same applies to the LD_V and LD_T operations.
+
+> ⚠️ VNNI combines elements from multiple rows of a single column, packing them into 32-bit values.
+It does not transform or modify the actual data, it only changes the packing format. 
+
+### Data Distribution (Packed Copies)
+
+When using MMA operations with **U8** or **U4** data types, such as `XE_8x16x32_S32S8S8S32_TT`, 
+the data for the A matrix must be packed before being consumed by the MMA instructions. In these cases, 
+a packed load operation must be used.
+
+```c++
+XE_2D_Packed_<DataType>x<BlockHeight>x<BlockWidth>_<LD|ST>_<N|T>
+```
+Example for `XE_2D_Packed_U8x32x32_LD_N` which loads a 32 rows by 32 columns block of elements, 
+each **work-item gets two adjacent columns**, each containing **32 elements**
+
+| Work-item | Columns Loaded |
+|-----------|----------------|
+| 0         | Column 0, 1    |
+| 1         | Column 2, 3    |
+| 2         | Column 4, 5    |
+| ...       | ...            |
+| 15        | Column 30, 31  |
+
+---
+
+Note: The number of adjacent columns assigned to each work-item is based on the BlockWidth value 
+in the 2D block copy operation divided by the number of work-items (16). In this case, each work-item 
+loads 2 columns because BlockWidth is 32 columns. If the copy operation has 64 columns, then each 
+work-item will load 4 adjacent columns.
```

Comment:
I do not think this is true. I just tested XE_2D_Packed_U8x8x64_LD_N in copy_debug, which packs 2 columns. I believe the actual number of consecutive columns per workitem is "16 / sizeof_bits_v<T>"